pkg/controllers/tagging/tagging_controller.go

```diff
@@ -281,3 +327,15 @@ func (tc *Controller) enqueueNode(obj interface{}, action func(node *v1.Node) er
 	tc.workqueue.Add(item)
 	klog.Infof("Added %s to the workqueue", item)
 }
+
+func (tc *Controller) isTaggingRequired(node *v1.Node) bool {
+	if node.Labels != nil {
+		if labelValue, ok := node.Labels[taggingControllerLabelKey]; !ok || labelValue != taggingControllerLabelValue {
```

Comment:
This would mean if the `--tags` flag value ever changed, CCM wouldn't update the tags on existing nodes to match, correct? I'm wondering if the label value should be a hash of the tags that it ensured exists on the instance

Reply:
That's correct, yes. And thanks for the call out. I was hoping that these tags would stay the same and we would possibly add logic in the future to take these from a Config Map instead of flags to make them dynamic although I do agree that some customers might want to update whatever tags they're adding. Let me see if I can support that.

Reply:
@rifelpet that's a great idea :100:

Reply:
Updated the PR with this check. 😄